hazelcast/src/main/java/com/hazelcast/spi/impl/securestore/SecureStore.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.securestore;
+
+import com.hazelcast.nio.Disposable;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * A Secure Store abstraction for interacting with systems that manage symmetric
+ * encryption keys.
+ */
+public interface SecureStore extends Disposable {
```

Comment:
Minor: how come these classes are in OS?

Reply:
Well, it's not a minor remark, it's actually a valid question. The goal was to make the secure store abstraction independent of hot-restart. I could have placed the APIs in a different package in the hot-restart submodule but I thought this was cleaner. At any rate, the top-level EE module (where the actual secure store implementations live) needs access to the APIs from some lower-level layer (= either the hot-restart submodule or OS).

We do something similar in OS with `HotRestartService` (which is public in OS), so I thought a similar approach might work for the secure store APIs (especially if we keep them internal). But at the same time, it does not feel very nice to me. The best would be to have a common EE sub-module where we could put stuff like this. Or maybe I should just create a special non-hot-restart package in the hot-restart module, to keep the damage under control.

Reply:
Right, I would agree it would be best if we had a common EE sub-module but I don't think this is a big issue since the classes are private.